marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -147,11 +147,21 @@ handleQuery (ReceiveWebSocketMessage msg next) = do
               -- if this is the wallet's MarloweApp...
               if (plutusAppId == marloweAppId) then case runExcept $ decodeJSON $ unwrap rawJson of
                 Left decodingError -> addToast $ decodingErrorToast "Failed to parse contract update." decodingError
-                Right lastResult -> case lastResult of
-                  -- FIXME: unpack these updates properly and respond appropriately in each case
-                  OK -> addToast $ successToast "all okay"
-                  SomeError marloweError -> addToast $ errorToast "something went wrong" Nothing
-                  Unknown -> pure unit
+                Right lastResult -> do
+                  let
+                    lastMarloweAppEndpointCall = view _lastMarloweAppEndpointCall dashboardState
+                  handleAction $ DashboardAction $ Dashboard.SetLastMarloweAppEndpointCall Nothing
+                  case lastResult of
+                    OK -> case lastMarloweAppEndpointCall of
+                      Just Create -> addToast $ successToast "Contract initialised."
+                      Just ApplyInputs -> addToast $ successToast "Contract update applied."
+                      _ -> pure unit
+                    -- FIXME: unpack marloweError to provide more informative feedback
+                    SomeError marloweError -> case lastMarloweAppEndpointCall of
+                      Just Create -> addToast $ errorToast "Failed to initialise contract." Nothing
+                      Just ApplyInputs -> addToast $ errorToast "Failed to update contract." Nothing
```

Comment:
If someone else (@hrajchert, @jhbertra) has time to finish this before I'm back, this is the last thing to do. The `MarloweError` type contains comments explaining when and how each error can arise.